bluesky/callbacks/core.py

```diff
@@ -42,7 +42,13 @@ def f(*args, **kwargs):
 class CallbackBase:
     def __call__(self, name, doc):
         "Dispatch to methods expecting particular doc types."
-        return getattr(self, name)(doc)
```

Comment:
During our discussion on Slack where I encouraged this PR, I didn't realize that we were already returning the output of the method. How forward-looking of us! :-D

Given that, I don't understand why any modification is needed. Can't the subclass's methods return whatever they want, including `(name, doc)`. Adding something extra here just strikes me as potentially confusing magic.

Reply:
1. It seems to be more efficient to have the name handled at the top level, where we have actual access rather than in the lower level where we need to copy the method name again.
2. This means that every subclass needs to be a no-op for more than one document router to be put together. That seems a bit counter productive.

Reply:
1. You mean like:

```py
class ScienceDoer(CallbackBase):
    def start(self, doc):
        ...
        return 'start', new_doc
```

as opposed to

```py
class ScienceDoer(CallbackBase):
    def start(self, doc):
        ...
        return super().start(new_doc) .   # base class will return ('start', new_doc)
```

2. Can you give me a very succinct example?

Reply:
Returning from super might be ok.

```python
ls = LiveScatter(...)
cb2 = OtherCallback(...)
L = []
RE.subscribe(lambda *x: L.append(cb2(*ls(*x))))
```
Does not work if LiveScatter doesn't send out a stream of documents.

Reply:
> Returning from super might be ok.

Wait...I'm showing you how I think your PR would work, not an alternative. How else would you involve the base class other than to call `super()`?

Reply:
I'm illustrating that adding complexity to the base class does not actually make the subclass simpler. Sure you don't have to type the method name again, as you say, but I think you would need to call `super()` which is more verbose and, in my opinion, also more opaque / harder to follow. See what I mean? Am I wrong about what the subclass would work? Maybe you should type on your ideal version of `ScienceDoer`'s return statement.

Reply:
In this PR the subclass version of start returns a dict. That dict then gets associated with a name in the call method. Most of the document routers so far haven't overrided the call method.

```python
class ScienceDoer(CallbackBase):
    def start(self, doc):
        doc = doc.update(hello='world')
        return doc
```
This modifies the start document, it then ships out the new document, which when call finishes looks like a proper name, document pair

Reply:
What I was missing here is that the _caller_ of `start(doc)` is typically accessing it indirectly via `__call__(name, doc)`. It follows that no call to `super()` is needed: the subclass can just return the `doc` and rely on `__call__` to do the rest.